src/include/kernel/ext_manifest.h

```diff
@@ -57,9 +57,10 @@ enum ext_man_elem_type {
 
 /* EXT_MAN_ELEM_CONFIG_DATA elements identificators */
 enum config_elem_type {
+	EXT_MAN_CONFIG_EMPTY		= 0,
 	EXT_MAN_CONFIG_IPC_MSG_SIZE	= 1,
 	EXT_MAN_CONFIG_MEMORY_USAGE_SCAN = 2, /**< ABI3.18 */
-	EXT_MAN_CONFIG_LAST_ELEM,	/**< keep it at the end of enum list */
+	EXT_MAN_CONFIG_LAST_ELEM	= EXT_MAN_CONFIG_MEMORY_USAGE_SCAN, /**< keep up to date with last enum */
```

Comment:
Still not following why we need a "last" ? Surely if one side does not know about an enum then we just ignore.

Reply:
@lgirdwood it is used to defien the fixed length array of the elems in the config data. It isnt really needed in the kernel as the kernel can determine the number of elements based on the hdr->size and the elem size. 

/** EXT_MAN_ELEM_CONFIG_DATA elements (ABI3.17) */
struct ext_man_config_data {
        struct ext_man_elem_header hdr;

        struct config_elem elems[EXT_MAN_CONFIG_LAST_ELEM];
} __packed;

Reply:
@lgirdwood SOF firmware header need to define the array with LAST_ELEM. Shouldn't we keep this for kernel header also? For practically kernel header don't need LAST_ELEM. We can calculate it by hdr->size and the elem size.

Reply:
This will constantly need updating unless it's set at MAX_INT or we use hdr->size * elem->size.

Reply:
@lgirdwood I defined MAX macro. @ktrzcinx I chose 8 as initial MAX value for each extended manifest. Do you think 8 is good total number to start with?

Reply:
I don't like the idea of handling this as constant size array from kernel side - there's no reason for that, it can be easily handled as flex array so I don't see any pros of constant size array here, only drawback of synchronising this value.
@fredoh9 how would you like to handle older FW with lower number of elements (older one) after introducing explicit dictionary capacity in kernel?

From FW side, explicit defining capacity instead using last enum value has some pros (explicit solution) and drawbacks (manual updating size) so I think it can be done here, but as long as we use each key once or none at all, I don't see need for it. 

Explicit defining `EXT_MAN_CONFIG_EMPTY` is change in good direction.

Reply:
@ktrzcinx the only reason I'm moving to fixed size, there was concern about ABI change at every update. Kernel driver parses this with hdr information.